src/main/java/seedu/address/logic/commands/NearbyCommand.java

```diff
@@ -45,7 +46,18 @@ public CommandResult execute() throws CommandException {
         }
 
         model.updateSelectedPerson(nearbyList.get(targetIndex.getZeroBased()));
-        EventsCenter.getInstance().post(new JumpToNearbyListRequestEvent(targetIndex));
+        Index index;
+        switch (model.getCurrentList()) {
+        case "blacklist":
+            index = Index.fromZeroBased(model.getFilteredBlacklistedPersonList().indexOf(model.getSelectedPerson()));
+            break;
+        case "whitelist":
+            index = Index.fromZeroBased(model.getFilteredWhitelistedPersonList().indexOf(model.getSelectedPerson()));
+            break;
+        default:
+            index = Index.fromZeroBased(model.getFilteredPersonList().indexOf(model.getSelectedPerson()));
+        }
```

Comment:
Thanks for using this method for now, I'll push it up to the ListObserver class once im done with it.